servers/physics_2d/godot_area_2d.cpp

```diff
@@ -199,20 +201,13 @@ void GodotArea2D::set_monitorable(bool p_monitorable) {
 }
 
 void GodotArea2D::call_queries() {
-	if (monitor_callback_id.is_valid() && !monitored_bodies.is_empty()) {
+	if (monitor_callback.is_valid() && !monitored_bodies.is_empty()) {
 		Variant res[5];
 		Variant *resptr[5];
 		for (int i = 0; i < 5; i++) {
 			resptr[i] = &res[i];
 		}
 
-		Object *obj = ObjectDB::get_instance(monitor_callback_id);
-		if (!obj) {
-			monitored_bodies.clear();
-			monitor_callback_id = ObjectID();
-			return;
-		}
-
```

Comment:
If the function returns in the monitored bodies section, then the monitored areas section is not executed. I made it so it always checks for both bodies and areas

Reply:
Yes indeed, the return seems to be wrong, it should use an else case instead. Good catch :)

Although there should be still two different cases to handle:
- If the callback is null or there's no overlapping bodies, just do nothing.
- If the callback is not valid anymore, it should also be set to null to avoid more processing.

Reply:
Both cases are considered in the last changes. I guess by setting `monitor_callback = Callable()` will make `monitor_callback.is_null()` to be true